src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -911,6 +999,65 @@ CHIP_ERROR ParseAttributePath(jobject attributePath, EndpointId & outEndpointId,
     return CHIP_NO_ERROR;
 }
 
+/**
+ * Takes objects in eventPathList, converts them to app:EventPathParams, and appends them to outEventPathParamsList.
+ */
+CHIP_ERROR ParseEventPathList(jobject eventPathList, std::vector<app::EventPathParams> & outEventPathParamsList)
+{
+    jint listSize;
+    ReturnErrorOnFailure(JniReferences::GetInstance().GetListSize(eventPathList, listSize));
+
+    for (uint8_t i = 0; i < listSize; i++)
+    {
+        jobject eventPathItem = nullptr;
+        ReturnErrorOnFailure(JniReferences::GetInstance().GetListItem(eventPathList, i, eventPathItem));
+
+        EndpointId endpointId;
+        ClusterId clusterId;
+        EventId eventId;
+        ReturnErrorOnFailure(ParseEventPath(eventPathItem, endpointId, clusterId, eventId));
+        outEventPathParamsList.push_back(app::EventPathParams(endpointId, clusterId, eventId));
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ParseEventPath(jobject eventPath, EndpointId & outEndpointId, ClusterId & outClusterId, EventId & outEventId)
+{
+    JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
+
+    jmethodID getEndpointIdMethod = nullptr;
+    jmethodID getClusterIdMethod  = nullptr;
+    jmethodID getEventIdMethod    = nullptr;
+
+    ReturnErrorOnFailure(JniReferences::GetInstance().FindMethod(
+        env, eventPath, "getEndpointId", "()Lchip/devicecontroller/model/ChipPathId;", &getEndpointIdMethod));
+    ReturnErrorOnFailure(JniReferences::GetInstance().FindMethod(
+        env, eventPath, "getClusterId", "()Lchip/devicecontroller/model/ChipPathId;", &getClusterIdMethod));
+    ReturnErrorOnFailure(JniReferences::GetInstance().FindMethod(env, eventPath, "getEventId",
+                                                                 "()Lchip/devicecontroller/model/ChipPathId;", &getEventIdMethod));
+
+    jobject endpointIdObj = env->CallObjectMethod(eventPath, getEndpointIdMethod);
+    VerifyOrReturnError(endpointIdObj != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    jobject clusterIdObj = env->CallObjectMethod(eventPath, getClusterIdMethod);
+    VerifyOrReturnError(clusterIdObj != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    jobject eventIdObj = env->CallObjectMethod(eventPath, getEventIdMethod);
+    VerifyOrReturnError(eventIdObj != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    uint32_t endpointId = 0;
```

Comment:
endpointId is uint16, please fix it

Reply:
I also know that. But, Java interger can cast to uint32_t. (You can check GetChipPathIdValue method.)
In 1054 line, endpointId value cast to EndpointId type(uint16_t).